chia/plotters/plotters.py

```diff
@@ -0,0 +1,346 @@
+                default=256,
+            )
+        if option is Options.MADMAX_WAITFORCOPY:
+            parser.add_argument(
+                "-w",
+                "--waitforcopy",
+                type=bool,
+                help="Wait for copy to start next plot",
+                default=True,
+            )
+        if option is Options.MADMAX_TMPTOGGLE:
+            parser.add_argument(
+                "-G",
+                "--tmptoggle",
+                help="Alternate tmpdir/tmpdir2 (default = false)",
+                default=False,
+            )
+        if option is Options.POOLCONTRACT:
+            parser.add_argument(
+                "-c",
+                "--contract",
+                type=str,
+                help="Pool Contract Address (64 chars)",
+                default="",
+            )
+        if option is Options.MADMAX_RMULTI2:
+            parser.add_argument(
+                "-K",
+                "--rmulti2",
+                type=binascii.unhexlify,
+                help="Thread multiplier for P2 (default = 1)",
+                default=1,
+            )
+        if option is Options.POOLKEY:
+            parser.add_argument(
+                "-p",
+                "--pool-key",
+                type=binascii.unhexlify,
+                help="Pool Public Key (48 bytes)",
+                default="",
+            )
+        if option is Options.FARMERKEY:
+            parser.add_argument(
+                "-f",
+                "--farmerkey",
+                type=binascii.unhexlify,
+                help="Farmer Public Key (48 bytes)",
+                default="",
+            )
+        if option is Options.BLADEBIT_WARMSTART:
+            parser.add_argument(
+                "-w",
+                "--warmstart",
+                type=bool,
+                help="Warm start (bool)",
+                default=False,
+            )
+        if option is Options.BLADEBIT_NONUMA:
+            parser.add_argument(
+                "-m",
+                "--nonuma",
+                type=bool,
+                help="Disable numa",
+                default=False,
+            )
+        if option is Options.VERBOSE:
+            parser.add_argument(
+                "-v",
+                "--verbose",
+                type=bool,
+                help="Set verbose",
+                default=False,
+            )
+        if option is Options.OVERRIDE_K:
+            parser.add_argument(
+                "--override",
+                type=bool,
+                help="Force size smaller than 32",
+                default=False,
+            )
+        if option is Options.ALT_FINGERPRINT:
+            parser.add_argument(
+                "-a",
+                "--alt_fingerprint",
+                type=int,
+                default=None,
+                help="Enter the alternative fingerprint of the key you want to use",
+            )
+        if option is Options.CONNECT_TO_DAEMON:
```

Comment:
TL;DR we don't need to expose this option. It was added as a hidden option just to support the GUI when a keyring passphrase is set. Users aren't likely to know when this option should be used.


The user experience when using the new passphrase-protected keyring is that CLI operations will prompt for the keyring passphrase to access keys. So running `chia plotters chiapos ...` without the `--connect_to_daemon` option will cause the keyring code to prompt for the user's passphrase. When plotting from the GUI, the user is asked to unlock their keyring at launch, and then the daemon maintains access to the keyring for the rest of the session. So when plotting from the GUI, we programmatically set the connect to daemon option '-D' so that the passphrase doesn't need to be re-entered.